docs/experimental/ticket-auction.md

```diff
@@ -114,17 +153,34 @@ $ tezos-client originate contract nft-auction transferring 0 from alice \
 
 New contract KT1HWaMyNmjVGMBPUSm3QxJnFRLi4LQJi1tG originated.
 ```
+
+Save auction contract address in an environment variable.
+
+```sh
+AUCTION_ADDRESS=KT1HWaMyNmjVGMBPUSm3QxJnFRLi4LQJi1tG
+```
+
 ## Demo
 Now that the wallet and auction contracts are originated Alice can create an NFT and auction it off. In the auction, no one will purchase the item at Alice's `opening_price` of 100tz so she drops the price to 90tz and Bob purchases the item.
 
 ### Mint NFT
 
 Alice mints herself a ticket based nft with metadata
 
+Following [TZIP-12 Token Metadata](https://gitlab.com/tzip/tzip/-/blob/master/proposals/tzip-12/tzip-12.md#token-metadata) we initialize the map of type `(map string bytes)`. The empty string `\"\"` key has a value of a TZIP-16 URI which points to a JSON representation of the token metadata.
+
+First, we will save the URI as a variable, and byte encode it. **Note, the URI I use just points to [this](https://github.com/tqtezos/ticket-tutorials/tree/main/tutorials/auction/ligo) repo. To actually follow TZIP-12 standard, a production level implementation should point to an actual [TZIP-16 Metadata JSON](https://gitlab.com/tzip/tzip/-/blob/master/proposals/tzip-16/tzip-16.md#metadata-json-format).  
+
+```sh
+$ URI=https://github.com/tqtezos/ticket-tutorials/tree/main/tutorials/auction/ligo
+# Encode the URI and format it
+$ URI=$(echo -n $URI | od -A n -t x1 | sed 's/ *//g'| tr -d '\n')
+```
```

Comment:
I think this can be spelled out in a bit shorter/readable way like so:
```
tobytes() {echo "0x$(echo -n $1 | od -A n -t x1 | tr -d ' \n')"}
TOKEN_METADATA="{\
Elt \"decimals\" $(tobytes 0); \
Elt \"ipfs_cid\" $(tobytes Qmb1s5K234gpBcmFFDnZBcufJpAWb8AtAhjf1fUH4z5f72); \
Elt \"name\" $(tobytes 'Demo NFT') \
}"
```
- roll outputting `0x` prefix into `tobytes` so that it doesn't have to be repeated
- use `tr` to remove both spaces and new lines (no sed)
- remove some superfluous quoting
- format metadata definition across multiple lines to make name-value structure apparent so that there's no need to duplicate names as env variables

Reply:
also, maybe 
```
"uri": ipfs://Qmb1s5K234gpBcmFFDnZBcufJpAWb8AtAhjf1fUH4z5f72
```
as an example instead of 
```
ipfs_cid: Qmb1s5K234gpBcmFFDnZBcufJpAWb8AtAhjf1fUH4z5f72
```
to better indicate that ipfs use here is just that - an example

Reply:
Added new tobytes function and kept `ipfs_cid` key as is after discussing offline.